test/functional/rpc_getblockstats.py

```diff
@@ -177,10 +149,11 @@ def run_test(self):
 
         self.log.info('Test tip including OP_RETURN')
         tip_stats = self.nodes[0].getblockstats(tip)
-        assert_equal(tip_stats["utxo_increase"], 6)
-        assert_equal(tip_stats["utxo_size_inc"], 441)
-        assert_equal(tip_stats["utxo_increase_actual"], 4)
-        assert_equal(tip_stats["utxo_size_inc_actual"], 300)
+        expected_tip_stats = self.expected_stats[self.max_stat_pos]
```

Comment:
https://github.com/bitcoin/bitcoin/pull/33184/commits/1ffdf5253388af3cf274efd723b7129d9639da8c

This is basically testing that two results of `getblockstats` match each other. It doesn't actually determine the `expected_tip_stats` in the test and then compare against the actual result of `getblockstats`.

Reply:
I also want to note that the generated blocks already contain `OP_RETURN` outputs  in coinbase transactions, so we don't need to create a specific `OP_RETURN` output in the test.

This might also remove the need to have a second block at all, since there's no need to create a new tip for the `OP_RETURN` test

Reply:
> This is basically testing that two results of `getblockstats` match each other. It doesn't actually determine the `expected_tip_stats` in the test and then compare against the actual result of `getblockstats`.

This is a good observation. I've reverted to using hardcoded expected values instead of circular getblockstats comparisons, ensuring the test validates actual correctness rather than just consistency.